### PR TITLE
Identify side inputs by (PTransform id, local name, PCollection id)

### PR DESCRIPTION
**Pull request by bsidhom**
This full triple is necessary to correctly identify side input views runner-side while serving ExecutableStages.

### CODE REVIEW

**Comment by bsidhom** [comment]
Note that `org.apache.beam.runners.core.construction.graph.GreedyPipelineFuserTest#sideInputRootsNewStage()` now fails.

**Comment by bsidhom** [comment]
@tgroh @lukecwik Is this roughly the approach we want to take? I've somehow broken one of the GreedyPipelineFuser tests, so I'm afraid that fusion might actually be broken.

I'm also not sure where `SideInputId` should live or exactly how it should be structured and reference from the Java side. It's a little strange because we have the notion of a PCollection id and currently assume this is equivalent to a side input. Unfortunately, a side input requires a full view reference for identification, so we need more information than just PCollection id. This ended up touching more places than I would have preferred.

File: model/pipeline/src/main/proto/beam_runner_api.proto [hunk 1]
```diff
@@ -719,6 +716,18 @@ message TimestampTransform {
...
 
+// A reference to a side input id.
+message SideInputId {
+  // The id of the PTransform that references this side input.
+  string transform_id = 1;
+
+  // The local name of this side input from the PTransform that references it.
+  string local_name = 2;
+
+  // The id of the PCollection that backs this side input.
```
Comment (tgroh) [inline model/pipeline/src/main/proto/beam_runner_api.proto]: I don't think you actually need this field - you can just retrieve it from the `ExecutableStage` components from https://github.com/apache/beam/pull/4948, and get the global name of the PCollection from the input PTransform

Comment (bsidhom) [reply model/pipeline/src/main/proto/beam_runner_api.proto]: Ah, good point.

**Comment by bsidhom** [comment]
@tgroh The tests pass now. Does this look roughly correct?

**Pull request merged**